lusid-sdk-python/tests/test_aggregation.py

```diff
@@ -0,0 +1,212 @@
+        tran_requests.append(test_utility.build_cash_funds_in_transaction_request(units=51501.0,
+                                                                                  currency=currency,
+                                                                                  trade_date=self.effective_date))
+        # create the transaction requests
+        idx = 0
+        for instrument in self.sorted_instrument_ids:
+            idx = idx + 1
+            tran_requests.append(test_utility.build_transaction_request(instrument_id=instrument,
+                                                                        units=100.0,
+                                                                        price=100.0 + idx,
+                                                                        currency=currency,
+                                                                        trade_date=self.effective_date,
+                                                                        transaction_type="Buy"))
+
+        response = self.run_aggregation(create_transaction_requests=tran_requests)
+        # add in some error checks here?
+        assert len(response) > 0
+        assert response[0]["Sum(" + self.AGGREGATION_KEY + ")"] == 10000.0             # Barclays
+        assert response[1]["Sum(" + self.AGGREGATION_KEY + ")"] == 20000.0             # National Grid
+        assert response[2]["Sum(" + self.AGGREGATION_KEY + ")"] == 30000.0             # Sainsburys
+
+    def run_aggregation(self, create_transaction_requests):
+
+        uuid_gen = uuid.uuid4()
+        scope = "finbourne"
+
+        # build the create portfolio request
+        original_portfolio_id = "Id-" + str(uuid_gen)
+        request = models.CreateTransactionPortfolioRequest("Portfolio-" + str(uuid_gen),
+                                                           original_portfolio_id,
+                                                           base_currency="GBP",
+                                                           created=self.effective_date)
+        # create the portfolio
+        portfolio_response = self.client.create_portfolio(scope, request)
+
+        self.assertEqual(portfolio_response.id.code, request.code)
+
+        portfolio_id = portfolio_response.id.code
+
+        # build the transaction requests
+        tran_req = []
+        for i in create_transaction_requests:
+            tran_req.append(i)
+
+        # upload the transactions to LUSID
+        upsert_response = self.client.upsert_transactions(scope, portfolio_id, tran_req)
+        assert len(upsert_response.links) > 0
+        # set up the prices used for the aggregation in the analytic stores
+        # is there a store in the list on the effective date?
+        analytic_stores = self.client.list_analytic_stores().values
+        count = 0
+        for store in analytic_stores:
+            if store.date_property == self.effective_date:
+                count = count + 1
```

Comment:
A more elegant solution might be to change this to a boolean instead of a count e.g. already_exists = True

Reply:
done